repose-aggregator/core/repose-core-api/src/test/scala/org/openrepose/core/systemmodel/config/AndTest.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package org.openrepose.core.systemmodel.config
+
+import javax.servlet.ServletInputStream
+import org.junit.runner.RunWith
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{FunSpec, Matchers}
+import org.springframework.mock.web.MockHttpServletRequest
+
+@RunWith(classOf[JUnitRunner])
+class AndTest extends FunSpec with Matchers with MockitoSugar {
+  describe("evaluate") {
+    it("should return true when all of the sub criteria are true") {
+      val filterCriteriaOne = mock[FilterCriterion]
+      when(filterCriteriaOne.evaluate(any(classOf[HttpServletRequestWrapper]))).thenReturn(true)
```

Comment:
It seems like it would have been easier to just write a test implementation of `FilterCriterion` that takes a boolean. You could have used it here and in several other tests instead of mocking.

Reply:
Could have. Could have also used anonymous classes that returned the desired value. My brain was just using mocks that day I guess.